src/clients/python/samples/basic/main.py

```diff
@@ -0,0 +1,49 @@
+import os
+
+import tigerbeetle_python as tb
+
+client = tb.ClientSync(cluster_id=0, replica_addresses=os.getenv("TB_ADDRESS", "3000"))
```

Comment:
I'd expect to see

```
with tb.ClientSync(cluster_id=0, replica_addresses=os.getenv("TB_ADDRESS", "3000")) as client:
   ...
```

Reply:
We could make the clients work as context managers too, but I don't think this is the behavior we want to encourage! We'd rather have the users create a single client somewhere and use that everywhere throughout their application.

Otherwise, they're very likely going to be creating / destroying clients per request. Destroying is not a problem, but the create would lead to a register and extra latency (which pending my message bus tweaks is _way_ higher than it should be!)

Reply:
🤔 but it feels like, if we have `deinit` method, we should encourage that it gets called, no? The fact that our own sample doesn't close the client is concerning (so, if you copy-paste this into one thousand tests, you'll leak one thousands C things, which is not good). 

I am not sure that `with` necessary encourages short scopes? I'd assume the user would have a `with tb.Client`  in main, and then pass the client everywhere as a parameter. 

Client is a resource. Idiomatic Python construct for dealing with resources, both short-lived and long-lived, is `with`. While it _could_ encourage some misuse, I think it's better to stick to language idioms, rather than preventively outthink the user?

We do use equivalent idioms (using in C#, try-with-resources in Java) for other languages.